lib/node_modules/@stdlib/blas/ext/join/docs/types/test.ts

```diff
@@ -0,0 +1,220 @@
+import join = require( './index' );
+
+
+// TESTS //
+
+// The function returns an ndarray...
+{
+	const x = zeros( [ 2, 2 ], {
+		'dtype': 'float64'
+	});
+
+	join( x, ',' ); // $ExpectType OutputArray
+	join( x, ',', {} ); // $ExpectType OutputArray
+}
+
+// The compiler throws an error if the function is provided a first argument which is not an ndarray...
+{
+	join( '5', ',' ); // $ExpectError
+	join( 5, ',' ); // $ExpectError
+	join( true, ',' ); // $ExpectError
+	join( false, ',' ); // $ExpectError
+	join( null, ',' ); // $ExpectError
+	join( void 0, ',' ); // $ExpectError
+	join( {}, ',' ); // $ExpectError
+	join( ( x: number ): number => x, ',' ); // $ExpectError
+
+	join( '5', ',', {} ); // $ExpectError
+	join( 5, ',', {} ); // $ExpectError
+	join( true, ',', {} ); // $ExpectError
+	join( false, ',', {} ); // $ExpectError
+	join( null, ',', {} ); // $ExpectError
+	join( void 0, ',', {} ); // $ExpectError
+	join( {}, ',', {} ); // $ExpectError
+	join( ( x: number ): number => x, ',', {} ); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided an options argument which is not an object...
+{
+	const x = zeros( [ 2, 2 ], {
+		'dtype': 'float64'
+	});
+
+	join( x, ',', '5' ); // $ExpectError
+	join( x, ',', true ); // $ExpectError
+	join( x, ',', false ); // $ExpectError
+	join( x, ',', [] ); // $ExpectError
+	join( x, ',', ( x: number ): number => x ); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided an invalid `dim` option...
+{
+	const x = zeros( [ 2, 2 ], {
+		'dtype': 'float64'
+	});
+
+	join( x, ',', { 'dim': '5' } ); // $ExpectError
+	join( x, ',', { 'dim': true } ); // $ExpectError
+	join( x, ',', { 'dim': false } ); // $ExpectError
+	join( x, ',', { 'dim': null } ); // $ExpectError
+	join( x, ',', { 'dim': [] } ); // $ExpectError
+	join( x, ',', { 'dim': {} } ); // $ExpectError
+	join( x, ',', { 'dim': ( x: number ): number => x } ); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided an invalid `keepdims` option...
+{
+	const x = zeros( [ 2, 2 ], {
+		'dtype': 'float64'
+	});
+
+	join( x, ',', { 'keepdims': '5' } ); // $ExpectError
+	join( x, ',', { 'keepdims': 5 } ); // $ExpectError
+	join( x, ',', { 'keepdims': null } ); // $ExpectError
+	join( x, ',', { 'keepdims': {} } ); // $ExpectError
+	join( x, ',', { 'keepdims': ( x: number ): number => x } ); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided an unsupported number of arguments...
+{
+	const x = zeros( [ 2, 2 ], {
+		'dtype': 'float64'
+	});
+
+	join(); // $ExpectError
+	join( x, ',', {}, {} ); // $ExpectError
+}
+
+// Attached to the function is an `assign` method which returns an ndarray...
+{
+	const x = zeros( [ 2, 2 ], {
+		'dtype': 'float64'
+	});
+	const y = empty( [], {
+		'dtype': 'generic'
+	});
+
+	join.assign( x, ',', y ); // $ExpectType genericndarray<number>
+	join.assign( x, ',', y, {} ); // $ExpectType genericndarray<number>
```

Comment:
```suggestion
	join.assign( x, ',', y ); // $ExpectType genericndarray<string>
	join.assign( x, ',', y, {} ); // $ExpectType genericndarray<string>
```

Returning a "generic" ndarray having a "number" subtype clearly does not make sense.